### PR TITLE
Set up base mixin with dynamic js language resource imports

### PR DESCRIPTION
**Pull request by BraydonKains**
We weren't using the base mixin yet, decided now would be a good time to get that started.

### CODE REVIEW

File: locales/en.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export const val = {
-	'en': { 'myLangTerm': 'I am a localized string!' }
+	'myLangTerm': 'I am a localized string!',
```
Comment (josephsaffron) [inline locales/en.js:2]: can probably kill the example one

File: src/components/d2l-teacher-course-creation-admin.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import { css, html, LitElement } from 'lit-element/lit-element.js';
-import { LocalizeMixin } from '@brightspace-ui/core/mixins/localize-mixin.js';
+import { css, html, LitElement } from 'lit-element/lit-element';
+import { BaseMixin } from '../mixins/base-mixin';
```
Comment (josephsaffron) [inline src/components/d2l-teacher-course-creation-admin.js:2]: Might as well do the rtl mixin too?

Comment (quinnbudan) [reply src/components/d2l-teacher-course-creation-admin.js:2]: The `BaseMixin` does the `RtlMixin` too already I think

File: src/components/d2l-teacher-course-creation-admin.js [hunk 3]
```diff
@@ -50,4 +33,4 @@ class TeacherCourseCreationAdmin extends LocalizeMixin(LitElement) {
 		`;
 	}
 }
-customElements.define('d2l-teacher-course-creation-admin', TeacherCourseCreationAdmin);
+customElements.define('d2l-tcc-admin', TeacherCourseCreationAdmin);
```
Comment (quinnbudan) [inline src/components/d2l-teacher-course-creation-admin.js:36]: I believe the tests are failing because of this

File: demo/admin.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+			import '../src/components/d2l-teacher-course-creation-admin.js';
+		</script>
+		<title>d2l-custom-teacher-course-creation</title>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta http-equiv="X-UA-Compatible" content="ie=edge">
+		<meta charset="UTF-8">
+	</head>
+	<body>
+		<d2l-demo-page page-title="d2l-custom-teacher-course-creation-admin">
+			<h2>d2l-custom-teacher-course-creation-admin</h2>
```
Comment (josephsaffron) [inline demo/admin.html:17]: nit: could change this to match.

**Pull request merged**